### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by goern**
Dockerfile putting ansible-container and all its dependencies in a container image...

### CODE REVIEW

**Comment by luebken** [comment]
I'm running into: 

``` 
$ docker run  --rm -v `pwd`:/work ansible-container --debug build
Error while fetching server API version: ('Connection aborted.', error(2, 'No such file or directory'))
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/cli.py", line 126, in commandline
    **vars(args))
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/engine.py", line 90, in cmdrun_build
    client = docker.AutoVersionClient(**client_kwargs)
  File "/usr/lib/python2.7/site-packages/docker/client.py", line 349, in __init__
    super(AutoVersionClient, self).__init__(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/docker/client.py", line 81, in __init__
    self._version = self._retrieve_server_version()
  File "/usr/lib/python2.7/site-packages/docker/client.py", line 105, in _retrieve_server_version
    'Error while fetching server API version: {0}'.format(e)
DockerException: Error while fetching server API version: ('Connection aborted.', error(2, 'No such file or directory'))
``` 

Running on Vagrant / Mac.

**Comment by luebken** [comment]
Next error: 
```
docker run --privileged  --rm -v `pwd`:/work -v /var/run/docker.sock:/var/run/docker.sock ansible-container --debug build
Ansible Container image has ID sha256:c6720f39b49fc5f5ede4545d6b28d26d991bf0ad054242e98ea253a75e98689d
Using temporary directory /tmp/tmpNeKopP...
Starting Compose engine to build your images...
Using temporary directory /tmp/tmpGPNN0P...
The DOCKER_CERT_PATH variable is not set. Defaulting to a blank string.
Cleaning up temporary directory /tmp/tmpGPNN0P...
Cleaning up temporary directory /tmp/tmpNeKopP...
Named volume "None:/usr/bin/docker:rw" is used in service "ansible-container" but no declaration was found in the volumes section.
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/cli.py", line 126, in commandline
    **vars(args))
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/engine.py", line 101, in cmdrun_build
    builder_img_id)
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/utils.py", line 189, in extract_hosts_touched_by_playbook
    builder_img_id=builder_img_id)
  File "/usr/lib/python2.7/site-packages/ansible_container-0.1-py2.7.egg/container/utils.py", line 163, in launch_docker_compose
    project = project_from_options('.', options)
  File "/usr/lib/python2.7/site-packages/compose/cli/command.py", line 31, in project_from_options
    environment=environment
  File "/usr/lib/python2.7/site-packages/compose/cli/command.py", line 69, in get_project
    config_data = config.load(config_details)
  File "/usr/lib/python2.7/site-packages/compose/config/config.py", line 322, in load
    match_named_volumes(service_dict, volumes)
  File "/usr/lib/python2.7/site-packages/compose/config/validation.py", line 73, in match_named_volumes
    volume_spec.repr(), service_dict.get('name')
ConfigurationError: Named volume "None:/usr/bin/docker:rw" is used in service "ansible-container" but no declaration was found in the volumes section.
```

Sounds like `ansible-container` looks for a local docker client when launching compose: 
https://github.com/ansible/ansible-container/blob/master/container/engine.py#L103

which it can't find when running inside a container:
https://github.com/ansible/ansible-container/blob/master/container/utils.py#L213

**Comment by goern** [comment]
You could mount /usr/bin/docker in via -v

**Comment by j00bar** [comment]
I'd probably prefer you export `DOCKER_*` environment variables into the container, esp. `DOCKER_HOST`. But you'd also have to export the value of `DOCKER_CERT_PATH` as a volume into the container as well.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM fedora
+
+MAINTAINER Christoph Görn <goern@redhat.com>
+
+RUN dnf install --setopt=tsflags=nodocs -y gcc redhat-rpm-config git python python-devel libffi-devel openssl-devel ansible && \
+    git clone https://github.com/goern/ansible-container && \
```
Comment (dustymabe) [inline Dockerfile]: why is this using goern/ansible-container ?

**Comment by chouseknecht** [comment]
If we choose to use pre-baked Ansible images, we'll use official ones. We had [these at one time](https://github.com/ansible/ansible-docker-base), but they have been deprecated. Now that we have ansible-container it might make sense to bring them back.

**Pull request closed without merging**